api/apps/geoprocessing/src/modules/protected-areas/worker/protected-area-processor.ts

```diff
@@ -1,60 +1,48 @@
 import { Injectable } from '@nestjs/common';
-import { Connection } from 'typeorm';
+import { EntityManager } from 'typeorm';
 import { Job } from 'bullmq';
+import { InjectEntityManager } from '@nestjs/typeorm';
+import { GeoJSON } from 'geojson';
+
+import { ShapefileService } from '@marxan/shapefile-converter';
 
 import { WorkerProcessor } from '../../worker';
 import { ProtectedAreasJobInput } from './worker-input';
-import { ShapefileService } from '@marxan/shapefile-converter';
 import { ProtectedArea } from '../protected-areas.geo.entity';
-import { GeoJSON } from 'geojson';
-import { GeometryExtractor } from './geometry-extractor';
 
 @Injectable()
 export class ProtectedAreaProcessor
   implements WorkerProcessor<ProtectedAreasJobInput, void> {
   constructor(
     private readonly shapefileService: ShapefileService,
-    private readonly geometryExtractor: GeometryExtractor,
-    private connection: Connection,
+    @InjectEntityManager() private readonly entityManager: EntityManager,
   ) {}
 
   async process(job: Job<ProtectedAreasJobInput, void>): Promise<void> {
-    const geo: GeoJSON = (
-      await this.shapefileService.transformToGeoJson(job.data.file)
-    ).data;
-
-    const geometries = this.geometryExtractor.extract(geo);
+    const { data: geo } = await this.shapefileService.transformToGeoJson(
+      job.data.file,
+    );
 
-    if (geometries.length === 0) {
-      throw new Error(
-        'No supported geometries found. Ensure that file contains FeatureCollection / Multipolygon / Polygon.',
-      );
-    }
-
-    const queryRunner = this.connection.createQueryRunner();
-
-    await queryRunner.connect();
-    await queryRunner.startTransaction();
-    try {
-      await queryRunner.manager.delete(ProtectedArea, {
+    return this.entityManager.transaction(async (transaction) => {
+      await transaction.delete(ProtectedArea, {
         projectId: job.data.projectId,
       });
-      await queryRunner.manager.insert(
-        ProtectedArea,
-        geometries.map((geometry) => ({
-          projectId: job.data.projectId,
-          fullName: job.data.file.filename,
-          theGeom: () =>
-            `st_multi(ST_GeomFromGeoJSON('${JSON.stringify(geometry)}'))`,
-        })),
-      );
 
-      await queryRunner.commitTransaction();
-    } catch (error) {
-      await queryRunner.rollbackTransaction();
-      throw error;
-    } finally {
-      await queryRunner.release();
-    }
+      await transaction.query(
+        `
+          INSERT INTO "wdpa"("the_geom", "project_id", "full_name")
```

Comment:
We are returning the inserted id, right?

Reply:
@aagm I haven't changed the flow right now, rather just fix the bug https://vizzuality.atlassian.net/browse/MARXAN-721 

If we need to return the IDs (assuming we cannot query them later) we should consider new issue with described use case.